update.ps1

```diff
@@ -1,201 +1,532 @@
+    }
+    
+    $createAccessTokenSplatParams = @{
+        Uri             = "https://login.microsoftonline.com/$($actionContext.Configuration.EntraIDTenantID)/oauth2/v2.0/token/"
+        Headers         = $headers
+        Method          = "POST"
+        ContentType     = "application/x-www-form-urlencoded"
+        UseBasicParsing = $true
+        Body            = $createAccessTokenBody
+        Verbose         = $false
+        ErrorAction     = "Stop"
+    }
+    
+    $createAccessTokenResonse = Invoke-RestMethod @createAccessTokenSplatParams
+    
+    Write-Verbose "Created access token. Result: $($createAccessTokenResonse | ConvertTo-Json)"
+    #endregion Create access token
+    
+    #region Create headers
+    $actionMessage = "creating headers"
+    
+    $headers = @{
+        "Authorization"   = "Bearer $($createAccessTokenResonse.access_token)"
+        "Accept"          = "application/json"
+        "Content-Type"    = "application/json;charset=utf-8"
+        "Mwp-Api-Version" = "1.0"
+    }
+    
+    Write-Verbose "Created headers. Result: $($headers | ConvertTo-Json)."
+    #endregion Create headers
+
+    #region Get account
+    # API docs: https://mwpapi.kpnwerkplek.com/index.html, specific API call: GET /api/users/{identifier}
+    $actionMessage = "querying account where [$($correlationField)] = [$($correlationValue)]"
+
+    $getKPNLisaAccountSplatParams = @{
+        Uri         = "$($actionContext.Configuration.MWPApiBaseUrl)/users/$($actionContext.References.Account)"
+        Method      = "GET"
+        Body        = @{
+            select = "$($accountPropertiesToQuery -join ',')"
+        }
+        Verbose     = $false
+        ErrorAction = "Stop"
     }
-    $PreviousPerson = Invoke-RestMethod @LisaRequest @SplatParams
 
-    $outputContext.PreviousData = $PreviousPerson | Select-Object -Property ([array] $outputContext.Data.PSObject.Properties.Name)
+    Write-Verbose "SplatParams: $($getKPNLisaAccountSplatParams | ConvertTo-Json)"
+
+    # Add header after printing splat
+    $getKPNLisaAccountSplatParams['Headers'] = $headers
+
+    $getKPNLisaAccountResponse = $null
+    $getKPNLisaAccountResponse = Invoke-RestMethod @getKPNLisaAccountSplatParams
+    $correlatedAccount = $getKPNLisaAccountResponse
+        
+    Write-Verbose "Queried account where [$($correlationField)] = [$($correlationValue)]. Result: $($correlatedAccount | ConvertTo-Json)"
+    #endregion Get account
+
+    #region Calulate action
+    $actionMessage = "calculating action"
+    if (($correlatedAccount | Measure-Object).count -eq 1) {
+        $actionMessage = "comparing current account to mapped properties"
+
+        # Set Previous data (if there are no changes between PreviousData and Data, HelloID will log "update finished with no changes")
+        $outputContext.PreviousData = $correlatedAccount.PsObject.Copy()
+
+        if ($actionContext.Configuration.updateManagerOnUpdate -eq $true) {
```

Comment:
Is dit niet iets wat je altijd zou willen?

Reply:
Nee, dit wil je als keuze hebben. Soms is de manager in HR niet (goed) gevuld en wil je dus de keuze hebben om dit verder met rust te laten.